### PR TITLE
adding UseClusterCA to BuildConfigSpec

### PR DESCRIPTION
**Pull request by coreydaley**

### CODE REVIEW

File: build/v1/types.go [hunk 1]
```diff
@@ -938,6 +938,10 @@ type BuildConfigSpec struct {
 	// When a BuildConfig is created, the 5 most recent failed builds are retained unless this value is set.
 	// If removed after the BuildConfig has been created, all failed builds are retained.
 	FailedBuildsHistoryLimit *int32 `json:"failedBuildsHistoryLimit,omitempty" protobuf:"varint,5,opt,name=failedBuildsHistoryLimit"`
+
+	// UseClusterCA denotes whether or not to use the cluster defined certificate authority
```
Comment (bparees) [inline build/v1/types.go:942]: // UseClusterCA denotes whether the builder container will use CAs from the cluster, or the CAs included in the builder image's own /etc/pki/ca-trust directory.  Setting this to true will overwrite the contents of /etc/pki/ca-trust w/ the CA trust values from the cluster.  Default: false

**Comment by bparees** [comment]
tweak to the godoc, otherwise lgtm.

/approve

will lgtm after we see a working PR that consumes the api (via a personal branch dep bump)

File: build/v1/consts.go [hunk 2]
```diff
@@ -197,4 +197,7 @@ const (
 
 	// ServiceCAKey is the ConfigMap key for the service signing certificate authority mounted into build pods.
 	ServiceCAKey = "service-ca.crt"
+
+	// MountsConfKey is the ConfigMap key for the build pods mounts.conf file
```
Comment (adambkaplan) [inline build/v1/consts.go:201]: nit:

```suggestion
	// MountsConfKey is the ConfigMap key for the build pod's container mount configuration file.
```

File: build/v1/types.go [hunk 3]
```diff
@@ -938,6 +938,11 @@ type BuildConfigSpec struct {
 	// When a BuildConfig is created, the 5 most recent failed builds are retained unless this value is set.
 	// If removed after the BuildConfig has been created, all failed builds are retained.
 	FailedBuildsHistoryLimit *int32 `json:"failedBuildsHistoryLimit,omitempty" protobuf:"varint,5,opt,name=failedBuildsHistoryLimit"`
+
+	// UseClusterCA denotes whether the builder container will use CAs from the cluster, or the CAs included in the builder image's own /etc/pki/ca-trust directory.
+	// Setting this to true will overwrite the contents of /etc/pki/ca-trust w/ the CA trust values from the cluster. Default: false
+	// +optional
+	UseClusterCA bool `json:"useClusterCA,omitempty" protobuf:"varint,6,opt,name=useClusterCA"`
```
Comment (adambkaplan) [inline build/v1/types.go:945]: Do we _ever_ want to support default and override values for this field?

If yes, then this needs to be a pointer so we can determine user intent.

Comment (adambkaplan) [inline build/v1/types.go:945]: Protobuf tag issue - I don't think this should have varint?

```suggestion
	UseClusterCA bool `json:"useClusterCA,omitempty" protobuf:"bool,6,opt,name=useClusterCA"`
```

**Review by adambkaplan — CHANGES_REQUESTED** [review]

File: build/v1/types.go (hunk 3 above)
Comment (coreydaley) [reply build/v1/types.go:945]: I checked with @bparees about this before doing it and he said it doesn't need to be a pointer.

Comment (coreydaley) [reply build/v1/types.go:945]: I copied this from another bool field in this same file, i am more than happy to change it, but was just following precedent.

Comment (coreydaley) [reply build/v1/types.go]: updated in current push

File: build/v1/consts.go (hunk 2 above)
Comment (coreydaley) [reply build/v1/consts.go]: updated in current push

File: build/v1/types.go (hunk 1 above)
Comment (coreydaley) [reply build/v1/types.go]: updated in current push

**Comment by bparees** [comment]
@coreydaley @adambkaplan regarding using a pointer, my expectation is that we would not add a cluster-wide override/default behavior for this, but if you think that is something you want the ability to add in the future then yes, use a pointer.

**Comment by bparees** [comment]
(I can see the argument for why someone might want a cluster wide default to use the cluster CAs while still retaining the user option to not use the cluster CAs, but i dunno how likely it is)

File: build/v1/types.go (hunk 3 above)
Comment (adambkaplan) [reply build/v1/types.go]: hrm...I don't see us using `protobuf"bool...` anywhere.

@deads2k @sttts is there a reason why we use `varint` for protobuf booleans?

Comment (sttts) [reply build/v1/types.go]: @smarterclayton defined the original proto for kube resources

**Comment by sttts** [comment]
Any progress here? Is this for 4.8?

**Comment by bparees** [comment]
> Any progress here? Is this for 4.8?

i think the bug this was addressing has been deferred to 4.8 at this point.

cc @gabemontero

**Pull request closed without merging**